crates/pypi-types/src/metadata.rs

```diff
@@ -210,6 +212,15 @@ impl Metadata23 {
             }
         }
 
+        // If dependencies are declared with Poetry, and `project.dependencies` is omitted, treat
+        // the dependencies as dynamic. The inclusion of a `project` table without defining
+        // `project.dependencies` is almost certainly an error.
```

Comment:
We should warn, that's a spec violation

Reply:
It's a spec violation in _Poetry_, right? What the user wrote is totally fine.

Reply:
Merging for now but will follow-up based on your reply.

Reply:
We should warn because using poetry <2 and `[project]` together without setting `dynamic` has unintended consequences, either on their own is technically compliant